endpoint.go

```diff
@@ -289,10 +290,14 @@ func (ep *endpoint) Join(sbox Sandbox, options ...EndpointOption) error {
 		return err
 	}
 
-	if err = sb.populateNetworkResources(ep); err != nil {
-		return err
+	sb.Lock()
```

Comment:
If you still need to update the heap (which is now done inside `populateNetworkResources()`), was wondering whether it makes sense to move and handle the osBox nil check inside `populateNetworkResources()`.

Also since on the leave path the nil check is handled inside `clearNetworkResources()`